tests/test_add_time_coords_to_idata.py

```diff
@@ -0,0 +1,525 @@
+        (
+            IDATA_WO_DATES,
+            "posterior_predictive",
+            "obs",
+            "obs_dim_0",
+            "1 day",
+            TypeError,
+        ),  # invalid time_step: string instead of timedelta
+        (
+            IDATA_WO_DATES,
+            "posterior_predictive",
+            "obs",
+            "obs_dim_0",
+            123,
+            TypeError,
+        ),  # invalid time_step: int instead of timedelta
+        (
+            IDATA_WO_DATES,
+            "posterior_predictive",
+            "obs",
+            "obs_dim_0",
+            timedelta(days=0),
+            ValueError,
+        ),  # invalid time_step: 0 days
+    ],
+)
+def test_input_types_add_coords(
+    idata, group, variable, dimension, time_step, expected_error
+):
+    """
+    Tests the validation of input types for
+    the add_time_coords_to_idata_dimension
+    function.
+    """
+
+    # check for expected error or
+    # for successful execution
+    if expected_error:
+        with pytest.raises(expected_error):
+            forecasttools.add_time_coords_to_idata_dimension(
+                idata=idata,
+                group=group,
+                variable=variable,
+                dimension=dimension,
+                start_date_iso=date(2022, 8, 1),
+                time_step=time_step,
+            )
+    else:
+        # if no error is expected,
+        # execute the function
+        old_dim = idata[group][variable][dimension].values
+        idata_out = forecasttools.add_time_coords_to_idata_dimension(
+            idata=idata,
+            group=group,
+            variable=variable,
+            dimension=dimension,
+            start_date_iso=date(2022, 8, 1),
+            time_step=time_step,
+        )
+
+        new_dim = idata_out[group][variable][dimension].values
+        assert not np.array_equal(old_dim, new_dim)
+        # validate that the function executed
+        # correctly and the dimension has been modified
+        assert dimension in idata_out.posterior_predictive[variable].coords
+        assert (
+            len(idata_out.posterior_predictive[variable].coords[dimension])
+            == idata_out.posterior_predictive[variable].sizes[dimension]
+        )
+
+
+def test_dates_add_time_coords_to_idata_dimension():
+    # set up test data
+    coords = {"obs_dim_0": [0, 1, 2, 3, 5]}
+    data = xr.Dataset(
+        {"obs": ("obs_dim_0", np.array([10, 20, 30, 40, 50]))}, coords=coords
+    )
+    idata = az.InferenceData(observed_data=data)
+
+    # test parameters
+    group = "observed_data"
+    variable = "obs"
+    dimension = "obs_dim_0"
+    start_date_iso = date(2024, 11, 19)
+    time_step = timedelta(days=2)
+
+    # expected output dates
+    expected_dates = np.array(
+        [
+            date(2024, 11, 19),
+            date(2024, 11, 21),
+            date(2024, 11, 23),
+            date(2024, 11, 25),
+            date(2024, 11, 27),
+        ]
+    ).astype("datetime64[D]")
```

Comment:
should this be 
```suggestion
    ).astype("datetime64[ns]")
```

Reply:
Making a note that when `xarry` time precision is relaxed, this will change back to `[D]`.